src/main/java/org/cyclops/cyclopscore/tileentity/CyclopsTileEntity.java

```diff
@@ -293,23 +287,25 @@ protected EnumFacing transformFacingForRotation(EnumFacing facing) {
 
     @Override
     public boolean hasCapability(@Nonnull Capability<?> capability, EnumFacing facing) {
-        return (sidedCapabilities != null && sidedCapabilities.contains(capability, transformFacingForRotation(facing)))
-                || (capabilities != null && capabilities.containsKey(capability))
-                || super.hasCapability(capability, facing);
+        if (facing != null){
+            if (sidedCapabilities.contains(capability, transformFacingForRotation(facing)))
+                return true;
+        }
+        return innerCapabilities.containsKey(capability) || super.hasCapability(capability, facing);
+
     }
 
     @Override
     @SuppressWarnings("unchecked")
     public <T> T getCapability(@Nonnull Capability<T> capability, EnumFacing facing) {
-        if (sidedCapabilities != null) {
-            Object value = sidedCapabilities.get(capability, transformFacingForRotation(facing));
-            if (value == null && facing != null) {
-                value = capabilities.get(capability);
-            }
-            if (value != null) {
-                return (T) value;
-            }
+        Object value;
+        if (facing != null){
+            value = sidedCapabilities.get(capability, facing);
         }
+        else value = innerCapabilities.get(capability);
```

Comment:
Remove `else`, because this is now inconsistent with `hasCapability`.

Reply:
i need the else there

Reply:
No. If value is still null at this point, you should query innerCapabilities.